### PR TITLE
gluon: Hide the definition of Partitioner

### PR DESCRIPTION
**Pull request by tylershunt**
No reason to keep it in a public header anymore (that I know of), the
only external reference to it is in `Distribution` where it's still a
friend.

Signed-off-by: Tyler Hunt <thunt@katanagraph.com>

### CODE REVIEW

**Comment by ddn0** [comment]
The truest definition of a friend: one who knows where you hide.

File: libgluon/src/Partition.cpp [hunk 1]
```diff
@@ -3442,16 +3416,19 @@ class Partitioner {
     Result<GraphTopology> topology_result =
         DoEdgeAssignment(pview, opt_.mirror_edges);
     if (!topology_result) {
-      GALOIS_LOG_FATAL("{}", topology_result.error());
+      GALOIS_LOG_FATAL("failed to assign edges: {}", topology_result.error());
```
Comment (roshandathathri) [inline libgluon/src/Partition.cpp:3419]: I think you meant ERROR

**Review by roshandathathri — APPROVED** [review]
Great!

File: libgluon/include/galois/graphs/DistGraph.h [hunk 2]
```diff
@@ -331,8 +331,9 @@ PartitionGraph(const DistOptions& opt) {
     }
   } else if (p_id == PartitionPolicy::kNotPartitioned || stat.num_hosts == 1) {
     assert(stat.num_hosts == 1);
-    Partitioner part(dist.get(), opt);
-    part.Partition();
+    if (auto res = Partition(dist.get(), opt); !res) {
+      return res.error();
```
Comment (gurbinder533) [inline libgluon/include/galois/graphs/DistGraph.h:335]: Why not `GALOIS_LOG_ERROR` here?

**Review by gurbinder533 — APPROVED** [review]
LGTM.

File: libgluon/include/galois/graphs/DistGraph.h (hunk 2 above)
Comment (roshandathathri) [reply libgluon/include/galois/graphs/DistGraph.h:335]: Logging at the source of the error is preferable typically

File: libgluon/src/Partition.cpp (hunk 1 above)
Comment (tylershunt) [reply libgluon/src/Partition.cpp:3419]: ah yep good catch.

File: libgluon/include/galois/graphs/DistGraph.h (hunk 2 above)
Comment (tylershunt) [reply libgluon/include/galois/graphs/DistGraph.h:335]: Yeah there's no great discipline  about where/when we log errors yet. In this case I tried to log every case inside `Partition` and avoided double notification here.

**Pull request merged**